File: examples/searchfreedesktops.clj
; Search the Free Rooms endpoint in Clojure
; Owain Kenway

; Make sure we have the right number of command line arguments.
(if (= (count *command-line-args*) 3)
  (def token (first *command-line-args*))
  ((.println System/err "Run with API key and start, end datetimes as arguments.")(System/exit 1))
)

; Import necessary classes.
(import uclapi.UCLApiConnection)
(import uclapi.ClusterRoom)
(import java.util.Hashtable)

; Put search parameters into a Hashtable.
(def params (Hashtable.))
(.put params "start_datetime" (second *command-line-args*))
(.put params "end_datetime" (.get *command-line-args* 2))

; Do our search and print results.
(println (clojure.string/join "\n" (ClusterRoom/searchAPI (UCLApiConnection. token) params)))

File: examples/searchfreedesktops.py
# UCL API example in Jython
# This example searches for free desktops in cluster rooms.

# Owain Kenway

import uclapi.UCLApiConnection
import uclapi.ClusterRoom
import java.util.Hashtable

import sys

# Get our key + query.
if len(sys.argv) != 4:
	sys.exit("Call with key and start/end dates as arguments")

key = sys.argv[1]
params = java.util.Hashtable()
params.put("start_datetime", sys.argv[2])
params.put("end_datetime", sys.argv[3])

# Make our connection and get our results.
conn = uclapi.UCLApiConnection(key)
results = uclapi.ClusterRoom.searchAPI(conn, params)

# Print them out.
for a in results:
	print(a.toString())
